src/Features/VisualBasic/Portable/ValidateFormatString/VisualBasicValidateFormatStringDiagnosticAnalyzer.vb

```diff
@@ -18,113 +15,45 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.ValidateFormatString
             Return VisualBasicSyntaxFactsService.Instance
         End Function
 
-        Friend Overrides Function TryGetArgsArgumentType(
-            semanticModel As SemanticModel,
-            arguments As SeparatedSyntaxList(Of SyntaxNode),
-            parameters As ImmutableArray(Of IParameterSymbol),
-            ByRef argsArgumentType As ITypeSymbol) As Boolean
-
-            Dim argsArgument As ArgumentSyntax = Nothing
-            If Not TryGetArgument(semanticModel, "args", arguments, parameters, argsArgument) Then
-                argsArgumentType = Nothing
-                Return False
-            End If
-
-            argsArgumentType = semanticModel.GetTypeInfo(argsArgument.GetArgumentExpression()).Type
-            Return True
+        Protected Overrides Function GetInvocationExpressionSyntaxKind() As SyntaxKind
+            Return SyntaxKind.InvocationExpression
         End Function
 
-        Protected Overrides Function TryGetFormatStringLiteralExpressionSyntax(
-            semanticModel As SemanticModel,
-            arguments As SeparatedSyntaxList(Of SyntaxNode),
-            parameters As ImmutableArray(Of IParameterSymbol),
-            ByRef formatStringLiteralExpressionSyntax As SyntaxNode) As Boolean
-
-            formatStringLiteralExpressionSyntax = Nothing
-
-            Dim formatArgumentSyntax As ArgumentSyntax = Nothing
-
-            If Not TryGetArgument(semanticModel, "format", arguments, parameters, formatArgumentSyntax) Then
-                Return False
-            End If
-
-            If Not formatArgumentSyntax.IsKind(SyntaxKind.SimpleArgument) Then
-                Return False
-            End If
-
-            Dim simpleArguementSyntax = DirectCast(formatArgumentSyntax, SimpleArgumentSyntax)
-
-            If simpleArguementSyntax.Expression Is Nothing Then
-                Return False
-            End If
+        Protected Overrides Function GetArgumentExpressionType(
+                semanticModel As SemanticModel,
+                argsArgument As SyntaxNode) As ITypeSymbol
 
-            If Not simpleArguementSyntax.Expression.IsKind(SyntaxKind.StringLiteralExpression) Then
-                Return False
-            End If
-
-            formatStringLiteralExpressionSyntax = DirectCast(simpleArguementSyntax.Expression,
-                LiteralExpressionSyntax)
-
-            Return True
+            Dim argumentSyntax = TryCast(argsArgument, ArgumentSyntax)
```

Comment:
You have `TryCast`, but don't check the result. Either make into a `DirectCast`, or check for failure if you think that's possible.

(The C# equivalent code does a direct cast).